src/main/scala/firrtl/passes/ReplSeqMem.scala

```diff
@@ -91,50 +88,35 @@ Optional Arguments:
 }
 
-class ReplSeqMem(transID: TransID) extends Transform with LazyLogging {
-  def execute(circuit:Circuit, map: AnnotationMap) = 
-    map get transID match {
-      case Some(p) => p get CircuitName(circuit.main) match {
-        case Some(ReplSeqMemAnnotation(t, _)) => {
-
-          val inputFileName = PassConfigUtil.getPassOptions(t).getOrElse(InputConfigFileName, "")
-          val inConfigFile = {
-            if (inputFileName.isEmpty) None 
-            else if (new java.io.File(inputFileName).exists) Some(new YamlFileReader(inputFileName))
-            else error("Input configuration file does not exist!")
-          }
-
-          val outConfigFile = new ConfWriter(PassConfigUtil.getPassOptions(t).get(OutputConfigFileName).get)
-          TransformResult(
-            (
-              Seq(
-                Legalize,
-                AnnotateMemMacros,
-                UpdateDuplicateMemMacros,
-                new AnnotateValidMemConfigs(inConfigFile),
-                new ReplaceMemMacros(outConfigFile),
-                RemoveEmpty,
-                CheckInitialization,
-                ResolveKinds,                                       // Must be run for the transform to work!
-                InferTypes,
-                ResolveGenders
-              ) foldLeft circuit
-            ) { 
-              (c, pass) =>
-                val x = Utils.time(pass.name)(pass run c)
-                logger debug x.serialize
-                x
-            } , 
-            None, 
-            Some(map)
-          )
-        }  
-        case _ => error("Unexpected transform annotation")
-      }
-      case _ => TransformResult(circuit, None, Some(map))
+class ReplSeqMem(transID: TransID) extends Transform with SimpleRun {
+  def passSeq(inConfigFile: Option[YamlFileReader], outConfigFile: ConfWriter) =
+    Seq(Legalize,
+        AnnotateMemMacros,
+        UpdateDuplicateMemMacros,
+        new AnnotateValidMemConfigs(inConfigFile),
+        new ReplaceMemMacros(outConfigFile),
+        RemoveEmpty,
+        CheckInitialization,
+        InferTypes,
+        ResolveKinds,         // Must be run for the transform to work!
+        ResolveGenders)
+
+  def execute(c: Circuit, map: AnnotationMap) = map get transID match {
+    case Some(p) => p get CircuitName(c.main) match {
+      case Some(ReplSeqMemAnnotation(t, _)) =>
+        val inputFileName = PassConfigUtil.getPassOptions(t).getOrElse(InputConfigFileName, "")
+        val inConfigFile = {
+          if (inputFileName.isEmpty) None 
+          else if (new File(inputFileName).exists) Some(new YamlFileReader(inputFileName))
+          else error("Input configuration file does not exist!")
+        }
+        val outConfigFile = new ConfWriter(PassConfigUtil.getPassOptions(t)(OutputConfigFileName))
+        run(c, passSeq(inConfigFile, outConfigFile))
+      case _ => error("Unexpected transform annotation")
     }
-}
\ No newline at end of file
+    case _ => TransformResult(c)
```

Comment:
What happened to map? How does that get passed along? [even if this transform doesn't change it]